tests/bluetooth/tester/src/btp_mcp.c

```diff
@@ -0,0 +1,1307 @@
+
+	bt_addr_le_copy(&ev.address, bt_conn_get_dst(conn));
+
+	ev.att_status = att_status;
+	ev.orders = sys_cpu_to_le16(orders);
+
+	tester_event(BTP_SERVICE_ID_MCP, BTP_MCP_PLAYING_ORDERS_SUPPORTED_EV, &ev, sizeof(ev));
+}
+
+static void btp_send_mcp_media_state_ev(struct bt_conn *conn, uint8_t att_status, uint8_t state)
+{
+	struct btp_mcp_media_state_ev ev;
+
+	bt_addr_le_copy(&ev.address, bt_conn_get_dst(conn));
+
+	ev.att_status = att_status;
+	ev.state = state;
+
+	tester_event(BTP_SERVICE_ID_MCP, BTP_MCP_MEDIA_STATE_EV, &ev, sizeof(ev));
+}
+
+static void btp_send_mcp_opcodes_supported_ev(struct bt_conn *conn, uint8_t att_status,
+					      uint32_t opcodes)
+{
+	struct btp_mcp_opcodes_supported_ev ev;
+
+	bt_addr_le_copy(&ev.address, bt_conn_get_dst(conn));
+
+	ev.att_status = att_status;
+	ev.opcodes = sys_cpu_to_le32(opcodes);
+
+	tester_event(BTP_SERVICE_ID_MCP, BTP_MCP_OPCODES_SUPPORTED_EV, &ev, sizeof(ev));
+}
+
+static void btp_send_mcp_content_control_id_ev(struct bt_conn *conn, uint8_t att_status,
+					       uint8_t ccid)
+{
+	struct btp_mcp_content_control_id_ev ev;
+
+	bt_addr_le_copy(&ev.address, bt_conn_get_dst(conn));
+
+	ev.att_status = att_status;
+	ev.ccid = ccid;
+
+	tester_event(BTP_SERVICE_ID_MCP, BTP_MCP_CONTENT_CONTROL_ID_EV, &ev, sizeof(ev));
+}
+
+static void btp_send_segments_obj_id_ev(struct bt_conn *conn, uint8_t att_status, uint64_t id)
+{
+	struct btp_mcp_segments_obj_id_ev ev;
+
+	bt_addr_le_copy(&ev.address, bt_conn_get_dst(conn));
+
+	ev.att_status = att_status;
+	ev.id = sys_cpu_to_le64(id);
+
+	tester_event(BTP_SERVICE_ID_MCP, BTP_MCP_SEGMENTS_OBJ_ID_EV, &ev, sizeof(ev));
+}
+
+static void btp_send_current_track_obj_id_ev(struct bt_conn *conn, uint8_t att_status, uint64_t id)
+{
+	struct btp_mcp_current_track_obj_id_ev ev;
+
+	bt_addr_le_copy(&ev.address, bt_conn_get_dst(conn));
+
+	ev.att_status = att_status;
+	ev.id = sys_cpu_to_le64(id);
+
+	tester_event(BTP_SERVICE_ID_MCP, BTP_MCP_CURRENT_TRACK_OBJ_ID_EV, &ev, sizeof(ev));
+}
+
+static void btp_send_media_cp_ev(struct bt_conn *conn, uint8_t att_status,
+				 const struct mpl_cmd *cmd)
+{
+	struct btp_mcp_media_cp_ev ev;
+
+	bt_addr_le_copy(&ev.address, bt_conn_get_dst(conn));
+
+	ev.att_status = att_status;
+	ev.opcode = cmd->opcode;
+	ev.use_param = cmd->use_param;
+	ev.param = sys_cpu_to_le32(cmd->param);
+
+	tester_event(BTP_SERVICE_ID_MCP, BTP_MCP_MEDIA_CP_EV, &ev, sizeof(ev));
+}
+
+static void btp_send_search_cp_ev(struct bt_conn *conn, uint8_t att_status,
+				  const struct mpl_search *search)
+{
+	struct btp_mcp_search_cp_ev *ev;
+	uint8_t param[SEARCH_LEN_MAX];
+
+	net_buf_simple_init(rx_ev_buf, 0);
+
+	ev = net_buf_simple_add(rx_ev_buf, sizeof(*ev));
+
+	bt_addr_le_copy(&ev->address, bt_conn_get_dst(conn));
+
+	ev->att_status = att_status;
+	ev->param_len = (uint8_t)search->search[0];
+
+	if (ev->param_len > (SEARCH_LEN_MAX - 1)) {
+		return;
+	}
+
+	ev->search_type = search->search[1];
+	strcpy(param, &search->search[2]);
```

Comment:
Are we sure that `param` is always NULL terminated?

Reply:
Fixed. Now we make sure it's NULL terminated before sending command to Search Control Point.